adobe/spectrum-web-components
# feat(icons-workflow): add support for Express workflow icons

## Description

Initial PR: https://github.com/adobe/spectrum-web-components/pull/3169
This PR is a follow up of the one above.  This new feature allows consumers to use the [Express icons](https://a4u-gateway.corp.adobe.com/bundles/62a0dcda6a689c08a6c091e4/overview) along with the Express theme variant by leveraging [conditional exports](https://nodejs.org/api/packages.html#conditional-exports).

To preview this feature you can go ahead and change the export conditions in storybook's `main.js` by adding **express** string as such:
```
exportConditions: ['browser', 'development', 'express'],
```


In a nutshell it:
- Adds `@adobe/spectrum-css-ccx-workflow-icons` which comes with Express icons.
- Modifies the build process of `icons-workflow` package to also process the Express icons.
- Adds entries in `icons-workflow/package.json` to specify where to find icons based on theme used (`express` vs `default`)


**Very important**: 
There may be some icons (5 icons at the moment) which are present in Express icons package but not in the Spectrum one. In this PR we are making available **only the icons that do have a Spectrum equivalent**. Rest will be ignored.

**Pretty important**: 
There are around ~800 Spectrum icons and around ~200 Express icons.
Therefore, since not all the Spectrum icons have an equivalent in Express, if you are using `express` export condition and you're looking for an icon that is not in the [Express icons package](https://a4u-gateway.corp.adobe.com/bundles/62a0dcda6a689c08a6c091e4/overview), you'll get the fallback, which is the Spectrum one.

## Related issue(s)



- https://github.com/adobe/spectrum-web-components/issues/2845

## Motivation and context


We should add built-in support for [Spectrum Express Workflow Icons](https://github.com/adobe/spectrum-css-workflow-icons/tree/main/dist/spectrum-css-ccx-workflow-icons). In this way we can continue to close the gap on full support for the Express variant of the Spectrum design language.

## How has this been tested?




-   [x] Toggling between "express" and "spectrum" icons
    1. Change the `exportConditions` of the storybook by appending "express" in the array: https://github.com/adobe/spectrum-web-components/blob/rocss/express-workflow-icons/storybook/main.js#L36
    2. Open storybook
    3. Compare it with https://opensource.adobe.com/spectrum-web-components/storybook/index.html?path=/story/icons-workflow--elements
    4. Observe there are icons such as `sp-icon-delete` which look different.  Indeed there are a lot which did not change... because there are ~800 spectrum and only ~200 express icons

## Screenshots (if appropriate)

## Types of changes




## Checklist




## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.
